crates/udd-sketch/src/lib.rs

```diff
@@ -292,7 +292,7 @@ impl UDDSketch {
             self.compact_buckets();
         }
 
-        for entry in other.buckets.iter() {
+        for entry in target.buckets.iter() {
```

Comment:
maybe we should just do `let other = other.clone()` above, so the original `other` is no longer reachable, should prevent bugs like this in the future 🤔

Reply:
Good point, made this change